### ISSUE TITLE
Size of s_string when static size set?

### ISSUE THREAD

**Issue by gwynevans** [jtpereyda/boofuzz#307]
I came across a scenario where I was trying to model a packet that had a calculated size then a fixed-size string buffer, e.g.
```
    s_initialize("example1")
    s_size("block", fuzzable=False, length=2)

    with s_block("block"):
        s_string("ABCD", fuzzable=False, size=32)
        s_byte(0)
```
This will currently generate a packet of the following form
`Transmitted 35 bytes: 05 00 41 42 43 44 00 00 ...
`where the overall size is correct but it seems to me that there's an argument that the size of `s_string` should be the `size` attribute (i.e. 32 dec/0x20) rather than the `len(value)` (4 dec)

To fix this could be done by overriding `__len__` in `string.py`, either in place or as a new primitive 
```
    def __len__(self):
        if self.size == -1:
            return len(self._value)
        else:
            return self.size
```
but I wanted to get an idea as to whether the current behaviour is intended or just an oversight?

**Comment by cq674350529** [comment]
I think it's an issue. In the `s_string()`, the `render()` will add padding to the value. But the `self._value` keeps unchanged, which will be used to calc the length.

**Comment by SR4ven** [comment]
Fixed by #422
